### PR TITLE
[Fix-11003]Task group queue is not updated to final state

### LINKED ISSUE #11003: [BUG-BE][TaskGroupOption] Task group queue is not updated to final state
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Task group queue is not updated to final state

image

### What you expected to happen

Task group queue is updated to final state

### How to reproduce

1. create workflow
2. bind task queue (queue size is 1)
3. start the workflow three times
4. Force a workflow to start  second time

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SnowMoon-Dev**
## Brief change log

close #11003 
ref [#1255](https://github.com/WhaleOps/WhaleScheduler/issues/1255)

## Verify this pull request

 Manually verified the change by testing locally.

image

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 1]
```diff
@@ -2989,7 +2988,7 @@ public TaskInstance releaseTaskGroup(TaskInstance taskInstance) {
             return null;
         }
         try {
-            while (taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
+            while (thisTaskGroupQueue.getForceStart() != 1 && taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
                 , thisTaskGroupQueue.getId(), TaskGroupQueueStatus.ACQUIRE_SUCCESS.getCode()) != 1) {
```
Comment (SnowMoon-Dev) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: Mainly modified here

Comment (SnowMoon-Dev) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: @lenboo @WangJPLeo

Comment (WangJPLeo) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: Excellent, you can add a desc of the modification reason in purpose.

**Review by WangJPLeo — DISMISSED** [review]

**Comment by caishunfeng** [comment]
I rerun the fail CI

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 2]
```diff
@@ -2989,7 +2988,7 @@ public TaskInstance releaseTaskGroup(TaskInstance taskInstance) {
             return null;
         }
         try {
-            while (taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
+            while (thisTaskGroupQueue.getForceStart() != 1 && taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]:
```suggestion
            while (thisTaskGroupQueue.getForceStart() == Flag.NO && taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
```

Comment (SnowMoon-Dev) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: ok thanks for the review suggestion, i have changed

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java (hunk 1 above)
Comment (SnowMoon-Dev) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: ok

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**